### PR TITLE
Knative Helm Chart

### PR DESCRIPTION
**Pull request by aquatribe**
**What this PR does / why we need it**:
This PR will create the community helm chart for the Knative Project that everyone can contribute to.

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by aquatribe** [comment]
cncf cla signed

File: incubator/knative/.helmignore [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Patterns to ignore when building packages.
```
Comment (davidkarlsen) [inline incubator/knative/.helmignore]: aim straight for the stable repo and avoid incubator

File: incubator/knative/templates/build/service.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+  name: {{ .Values.build.controller.name }}
+  namespace: {{ .Values.build.namespace }}
+spec:
+  ports:
+  - name: metrics
+    port: 9090
+    protocol: TCP
+    targetPort: 9090
```
Comment (davidkarlsen) [inline incubator/knative/templates/build/service.yaml]: using a named port is preferrable IMHO

File: incubator/knative/templates/eventing/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
+{{- if .Values.eventing.install -}}
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: eventing-controller
+  namespace: {{ .Values.eventing.namespace }}
```
Comment (davidkarlsen) [inline incubator/knative/templates/eventing/deployment.yaml]: afaik the app/chart/rel/heritage labels should be at this level

File: incubator/knative/templates/hooks/post-install.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+{{ if .Values.istio.install }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: post-installation-job
+  namespace: kube-system
```
Comment (davidkarlsen) [inline incubator/knative/templates/hooks/post-install.yaml]: do we need labels here too?

File: incubator/knative/templates/build/serviceaccount.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{{- if and (.Values.build.install) (not .Values.serving.install ) -}}
+{{ if .Values.serviceAccount.create }}
```
Comment (davidkarlsen) [inline incubator/knative/templates/build/serviceaccount.yaml]: .Values.serviceAccount.create does not exist - it seems to be enabled per item - so the conditional needs to follow that

File: incubator/knative/templates/eventing/clusterrole.yaml [hunk 6]
```diff
@@ -0,0 +1,69 @@
+{{- if .Values.eventing.install -}}
+{{ if .Values.rbac.enable }}
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```
Comment (davidkarlsen) [inline incubator/knative/templates/eventing/clusterrole.yaml]: split the different resources into separate files (cluster role, role binding, serviceaccount etc)

**Review by davidkarlsen — COMMENTED** [review]
Have a look at the comments. However this seems like a bit advanced install - so I'd like someone with more experience like @prydonius and @unguiculus to have a look as I'm quite fresh in the game

File: incubator/knative/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+- istio
+- serving
+- build
+sources:
+- https://github.com/knative/
+maintainers:
+- name: simplytunde
+  email: tunde@aquatribe.com
+- name: boluisa
+  email: babatunde@aquatribe.com
```
Comment (davidkarlsen) [inline incubator/knative/Chart.yaml]: since crd's are installed, you might want to require helm >= 2.10.0: https://github.com/helm/helm/releases

**Review by unguiculus — CHANGES_REQUESTED** [review]
Just a few remarks from quickly looking over it:
* Please compare with what you get with `helm create` with latest Helm and update accordingly.
* Make sure you read and apply our guidelines: https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md
* Please follow our RBAC best practices: https://github.com/helm/helm/blob/master/docs/chart_best_practices/rbac.md

**Comment by zcourts** [comment]
Is there any interest in moving this forward? 
We're planning to merge it in our fork but ideally would prefer it being upstream

**Comment by aquatribe** [comment]
@zcourts This will be prioritize for an update sometimes this weekend.

**Comment by gyliu513** [comment]
FYI @zxdiscovery @clyang82

File: incubator/knative/values.yaml [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+      pullPolicy: IfNotPresent
+    args:
+      - -logtostderr
+      - -stderrthreshold
+      - INFO
+
+  webhook:
+    replicaCount: 1
+    image:
+      repository: gcr.io/knative-releases/github.com/knative/eventing/cmd/webhook
```
Comment (zxDiscovery) [inline incubator/knative/values.yaml]: Is the content of this repository miss ``@sha256`` after ``gcr.io/knative-releases/github.com/knative/eventing/cmd/webhook`` ?

**Comment by tomleah** [comment]
@aquatribe any update on this?

**Comment by neeraj-laad** [comment]
@aquatribe Do you have any updates on this PR?

**Comment by gyliu513** [comment]
@aquatribe There are many different components in this chart, why not use subchart to organize the whole chart for better management?

File: incubator/knative/.helmignore (hunk 1 above)
Comment (simplytunde) [reply incubator/knative/.helmignore]: It has been moved to the stable repo

File: incubator/knative/templates/eventing/deployment.yaml (hunk 3 above)
Comment (simplytunde) [reply incubator/knative/templates/eventing/deployment.yaml]: We addressed this in the latest commit

File: incubator/knative/templates/hooks/post-install.yaml (hunk 4 above)
Comment (simplytunde) [reply incubator/knative/templates/hooks/post-install.yaml]: This was also addressed

File: incubator/knative/templates/build/serviceaccount.yaml (hunk 5 above)
Comment (simplytunde) [reply incubator/knative/templates/build/serviceaccount.yaml]: This has been fixed.

File: incubator/knative/templates/eventing/clusterrole.yaml (hunk 6 above)
Comment (simplytunde) [reply incubator/knative/templates/eventing/clusterrole.yaml]: Also addressed

File: incubator/knative/Chart.yaml (hunk 7 above)
Comment (simplytunde) [reply incubator/knative/Chart.yaml]: helmVersion requirement has been added. Thanks a lot @davidkarlsen for the review

**Pull request closed without merging**